### PR TITLE
#158911604 validate json request type

### PR DESCRIPTION
**Pull request by Dheavyman**
#### What does this PR do?
- Validate the incoming request to ensure it's content-type is application/json to 500 error
#### Description of Task to be completed?
- The task requires that the error thrown should be caught and handled with a descriptive error message. The solution ensures that such error is prevented from happening with a descriptive message to the client.
#### How should this be manually tested?
- Create a local copy with the following commands: `git fetch origin bg-fix-add-asset-json-158911604:bg-fix-add-asset-json-158911604`, then `git checkout bg-fix-add-asset-json-158911604`
- Start the application with the command: `python manage.py runserver`
- Using postman, send a request with a content-type of text or form-data to: `localhost:5000/api/v1/assets`
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[ #158911604](https://www.pivotaltracker.com/story/show/158911604)
#### Screenshots (if appropriate)
screen shot 2018-07-10 at 4 13 00 pm

#### Questions:
- N/A

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -137,6 +138,16 @@ def auth_header(generate_token=generate_token):
     }
 
 
+@pytest.fixture(scope='module')
+def auth_header_text(generate_token=generate_token):
+    return {
+        'Authorization': generate_token(),
+        'Content-Type': MIMETYPE_TEXT,
+        'Accept': MIMETYPE_TEXT
+
```
Comment (ibutiti) [inline tests/conftest.py]: Please remove the extra line

File: api/utilities/messages/error_messages.py [hunk 2]
```diff
@@ -19,7 +19,8 @@
     'category_not_found': 'This category does not exist in the database',
     'attribute_not_related': 'attribute with the id of {attribute_id} is not related to the asset category of id {asset_category_id}',  # noqa
     'invalid_id': 'Invalid id in parameter',
-    'key_error': '{} key not found'
+    'key_error': '{} key not found',
+    'json_type_required': 'Content-type should be application/json'
```
[the remark this answers is not in the document]
Comment (Dheavyman) [reply api/utilities/messages/error_messages.py]: @GbengaOshinaga Thanks for that.

File: tests/test_asset_endpoints.py [hunk 3]
```diff
@@ -168,3 +168,22 @@ def test_create_asset_with_unrelated_custom_attr(
...
+        asset["assetCategoryId"] = test_asset_category.id
+        asset["customAttributes"] = {"waranty": "expired", "length": "100"}
+        data = json.dumps(asset)
+        response = client.post(f'{api_v1_base_url}/assets',
+                               headers=auth_header_text, data=data)
+        response_json = json.loads(response.data.decode(CHARSET))
+
+        assert response.status_code == 400
+        assert response_json["status"] == "error"
+        assert response_json["message"] == "Content-type should be " \
```
Comment (ibutiti) [inline tests/test_asset_endpoints.py]: Correct this as @GbengaOshinaga said.

File: api/utilities/messages/error_messages.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Dheavyman) [reply api/utilities/messages/error_messages.py]: Done

File: tests/test_asset_endpoints.py (hunk 3 above)
Comment (ibutiti) [reply tests/test_asset_endpoints.py]: Also, for long strings, i think we can agree to use the `#noqa` flag or for pylint use the `#pylint: disable=<linter error code here>` flag

Comment (Dheavyman) [reply tests/test_asset_endpoints.py]: I personally like keeping to the max line of 80 as much as possible. The `#noqa` might be a nice workaround but we have to decide on the limit.

File: tests/conftest.py (hunk 1 above)
Comment (Dheavyman) [reply tests/conftest.py]: Done

**Comment by GbengaOshinaga** [comment]
@Dheavyman Squash your commits into one

File: api/views/asset.py [hunk 4]
```diff
@@ -20,6 +21,7 @@ class AssetResource(Resource):
     """
 
     @token_required
+    @validate_json_request
```
Comment (brayoh) [inline api/views/asset.py]: I think we should add this to all endpoints not jus the assets

Comment (Dheavyman) [reply api/views/asset.py]: @brayoh Yeah, but should I do that in this story?

Comment (brayoh) [reply api/views/asset.py]: Yes

Comment (brayoh) [reply api/views/asset.py]: the story doesn't seem to specify which endpoints so I assume it should be the entire codebase

Comment (Dheavyman) [reply api/views/asset.py]: Ok

**Comment by Dheavyman** [comment]
@brayoh All changes made, please review

File: api/views/asset.py (hunk 4 above)
Comment (Dheavyman) [reply api/views/asset.py]: Done, all endpoint now validate that the request is in json format

**Comment by brayoh** [comment]
Cool

**Pull request merged**